packages/common/src/utils/filters.ts

```diff
@@ -280,7 +283,10 @@ export const getFilterRuleWithDefaultValue = <T extends FilterRule>(
                         ? formatDate(
                               // Treat the date as UTC, then remove its timezone information before formatting
                               moment.utc(value).format('YYYY-MM-DD'),
-                              fieldTimeInterval, // Use the field's time interval if it has one
+                              // For QUARTER, we don't want to use the field's time interval(YYYY-[Q]Q) because the date is already in the correct format when generating the SQL
+                              fieldTimeInterval === TimeFrames.QUARTER
+                                  ? undefined
```

Comment:
when saving a filter, after adding it, it would change its formatting to "YYYY-[Q]Q" and save to the query. This should be treated as DAY, and the formatting should happen on the UI